pkg/auth/dependency/identity/loginid/provider.go

```diff
@@ -143,10 +143,45 @@ func (p *Provider) New(userID string, loginID loginid.LoginID) *Identity {
 	}
 }
 
+func (p *Provider) WithLoginID(iden *Identity, loginID loginid.LoginID) *Identity {
```

Comment:
There is quite a bit of duplicated logic in `New` and `WithLoginID`. Can you refactor it out (e.g. as a new function `populateLoginID`)?